frontend/src/containers/LibraryPage/LibraryPage.js

```diff
@@ -1,61 +1,180 @@
+    this.setState({
+      videoCards: transformedCards,
+      sortedVideoCardDates: sortedDates,
+    });
+  }
+
   toggleUploadModal = () => {
     const { showUploadModal } = this.state;
     this.setState({ showUploadModal: !showUploadModal });
-  }
+  };
 
   // eslint-disable-next-line no-unused-vars
   handleUpload = (addedVideos, videoNames) => {
     // [TM]: TODO WDB-104
-  }
+  };
 
   handleUploadModalClose = () => {
     this.toggleUploadModal();
-  }
+  };
 
   render() {
-    const { showUploadModal } = this.state;
+    const {
+      showUploadModal,
+      videoCards,
+      sortedVideoCardDates,
+      selectedCards,
+    } = this.state;
+
+    const handleSelect = (id) => {
+      if (selectedCards.find((cardId) => cardId === id)) {
+        const newSelectedCards = selectedCards.filter(
+          (cardId) => cardId !== id,
+        );
+        this.setState({ selectedCards: newSelectedCards });
+      } else {
+        this.setState({ selectedCards: [...selectedCards, id] });
+      }
+    };
 
     return (
-      <Grid className='root' container direction='column'>
+      <Grid
+        className="root"
+        style={{ height: videoCards.length === 0 ? '100vh' : 'auto' }}
+        container
+        direction="column"
+      >
         <UploadModal
           show={showUploadModal}
           onUpload={this.handleUpload}
           onClose={this.handleUploadModalClose}
         />
-        <Grid item>
-          <TopBar
-            title='Video Library'
-            onActionIconClick={() => {
-              /* [TM:] TODO WDB-29 */
-            }}
-            showAvatarAndLogout
-            firstName={window.sessionStorage.getItem('firstName')}
-            lastName={window.sessionStorage.getItem('lastName')}
-            iconsToShow={[UploadIcon]}
-            onIconsClick={[
-              this.toggleUploadModal,
-            ]}
-          />
-        </Grid>
-        <Grid item style={{ flexGrow: 1 }}>
-          <EmptyLibraryContent />
+        <Grid item className="flexGrow">
+          {selectedCards.length === 0 ? (
+            <TopBar
+              title="Video Library"
+              onActionIconClick={() => {
+                /* [TM:] TODO WDB-29 */
+              }}
+              showAvatarAndLogout
+              firstName={window.sessionStorage.getItem('firstName')}
+              lastName={window.sessionStorage.getItem('lastName')}
+              iconsToShow={[UploadIcon]}
+              onIconsClick={[this.toggleUploadModal]}
+            />
+          ) : (
+            <TopBar
+              title={`${selectedCards.length} ${
+                selectedCards.length === 1 ? 'video' : 'videos'
+              } selected`}
+              showArrow
+              onActionIconClick={() => this.setState({ selectedCards: [] })}
+            />
+          )}
         </Grid>
+        {videoCards.length !== 0 ? (
+          <Grid
+            className="cardContainer"
+            container
+            item
```

Comment:
I believe this was already asked, but are you supposed to add both item and a container to a grid?

Reply:
You can